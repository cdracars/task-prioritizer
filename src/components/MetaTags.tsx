import { Helmet } from 'react-helmet-async';

interface MetaTagsProps {
  title?: string;
  description?: string;
  canonicalUrl?: string;
}

/**
 * Component for managing meta tags to improve SEO
 */
const MetaTags: React.FC<MetaTagsProps> = ({
  title = 'Task Prioritizer - Organize Your Tasks Effectively',
  description = 'A simple app to help you prioritize your tasks through pairwise comparisons',
  canonicalUrl = 'https://taskprioritizer.app',
}) => {
  return (
    <Helmet>
      <title>{title}</title>
      <meta name="description" content={description} />
      <link rel="canonical" href={canonicalUrl} />

      {/* Open Graph / Facebook */}
      <meta property="og:type" content="website" />
      <meta property="og:url" content={canonicalUrl} />
      <meta property="og:title" content={title} />
      <meta property="og:description" content={description} />
      <meta property="og:image" content={`${canonicalUrl}/icons/logo512.png`} />

      {/* Twitter */}
      <meta property="twitter:card" content="summary_large_image" />
      <meta property="twitter:url" content={canonicalUrl} />
      <meta property="twitter:title" content={title} />
      <meta property="twitter:description" content={description} />
      <meta
        property="twitter:image"
        content={`${canonicalUrl}/icons/logo512.png`}
      />
    </Helmet>
  );
};

export default MetaTags;
