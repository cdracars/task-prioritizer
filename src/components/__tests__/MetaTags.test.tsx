import React from 'react';
import { render } from '@testing-library/react';
import { HelmetProvider } from 'react-helmet-async';
import MetaTags from '../MetaTags';

describe('MetaTags', () => {
  it('renders without crashing', () => {
    const { container } = render(
      <HelmetProvider>
        <MetaTags />
      </HelmetProvider>
    );
    expect(container).toBeDefined();
  });

  // Simple snapshot test since direct testing with helmet is complex
  it('matches snapshot with default props', () => {
    const { container } = render(
      <HelmetProvider>
        <MetaTags />
      </HelmetProvider>
    );
    expect(container).toMatchSnapshot();
  });

  it('matches snapshot with custom props', () => {
    const customProps = {
      title: 'Custom Title',
      description: 'Custom description for testing',
      canonicalUrl: 'https://example.com/test',
    };

    const { container } = render(
      <HelmetProvider>
        <MetaTags {...customProps} />
      </HelmetProvider>
    );
    expect(container).toMatchSnapshot();
  });
});
