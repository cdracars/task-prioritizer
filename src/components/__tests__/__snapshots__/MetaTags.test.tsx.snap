// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MetaTags matches snapshot with custom props 1`] = `<div />`;

exports[`MetaTags matches snapshot with default props 1`] = `<div />`;
